src/win/pipe.c

```diff
@@ -87,6 +87,16 @@ static void eof_timer_destroy(uv_pipe_t* pipe);
 static void eof_timer_close_cb(uv_handle_t* handle);
 
 
+/* If a single write is too big, the kernel may fail when calling WriteFile.
+ * TODO: libuv assumes that writes can't fail, so this needs to be handled at the application level
+ */
+static DWORD os_maximum(ULONG bytes) {
+  /* if (bytes > 0xFFFF)
+    return 0xFFFF; */
+  return (DWORD)bytes;
```

Comment:
This function is either pointless or wrong, depending on whether `sizeof(DWORD) <= sizeof(ULONG)`.

Reply:
DWORD === ULONG, so it's true it doesn't do much. but libuv assumes that writes can't fail, so limiting here didn't end up working.

Reply:
Also, this would be still a bit larger than the documented limit for XP-32 (the limit for XP-64 is smaller again by half). If I don't need to support that platform, then I think it'll be sufficient just to note that the users of libuv should limit the maximum size of their writes (to about 1 GB, from my brief testing on Windows). This limitation is mostly already in place due to the definition of `uv_buf_t` anyways (where MS used a ULONG to represent the size of memory).